src/recon.rs

```diff
@@ -1708,7 +1707,7 @@ unsafe fn read_coef_tree<BD: BitDepth>(
         let mut cf_ctx: u8 = 0;
         let eob;
         let cf: *mut BD::Coef;
-        let mut cbi: *mut CodedBlockInfo = 0 as *mut CodedBlockInfo;
+        let mut cbi = None;
```

Comment:
```suggestion
        let mut cbi = &mut Default::default();
```

`CodedBlockInfo` is small (10 bytes), so creating a `Default::default()` instead of a `None` has minimal overhead.  This way no `.unwrap()`s are needed.  A more complex solution tying together `frame_thread.pass` as an `enum` to the `CodedBlockInfo` works, too, but is more complex and probably equally efficient.